command/ls.go

```diff
@@ -127,19 +130,24 @@ func List(
 
 		if err := object.Err; err != nil {
 			merror = multierror.Append(merror, err)
+			printError(l.fullCommand, l.op, err)
 			continue
 		}
 
 		msg := ListMessage{
 			Object:           object,
-			showEtag:         showEtag,
-			showHumanized:    humanize,
-			showStorageClass: showStorageClass,
+			showEtag:         l.showEtag,
+			showHumanized:    l.humanize,
+			showStorageClass: l.showStorageClass,
 		}
 
 		log.Info(msg)
 	}
-
+	if merror != nil {
```

Comment:
Could not get why we need custom error with the conditional displayErr flag. If the purpose is not logging them at the end, you can simply remove [ExitErrHandler](https://github.com/peak/s5cmd/blob/master/command/app.go#L98-L101) callback and directly return  `merror` after each command.

Reply:
Thanks! Your suggestion is elegant way of doing this and its great. It seems we would need some extra care in order not to print errors repeatedly (e.g., print in current function and print again in calling function after return).  I will do modifications accordingly.